lib/spack/spack/package.py

```diff
@@ -2097,15 +2097,34 @@ def fetch_remote_versions(self):
             tty.die("Package.fetch_versions couldn't connect to:", e.url,
                     e.message)
 
+    @property
+    def rpath_deps(self):
+       """Return immediate or transitive RPATHs depending on the package."""
+       if self.transitive_rpaths:
+           return [d for d in self.spec.traverse(root=False, deptype=('link'))]
+       else:
+           return self.spec.dependencies(deptype='link')
+
+
     @property
     def rpath(self):
         """Get the rpath this package links with, as a list of paths."""
-        rpaths = [self.prefix.lib, self.prefix.lib64]
-        deps = self.spec.dependencies(deptype='link')
-        rpaths.extend(d.prefix.lib for d in deps
-                      if os.path.isdir(d.prefix.lib))
-        rpaths.extend(d.prefix.lib64 for d in deps
-                      if os.path.isdir(d.prefix.lib64))
+        rpaths = self.libs.directories
```

Comment:
@alalazo i guess here I need to go through `self.spec.libs`? But python complains that `Spec` has no attribute `libs`. I know it's my misunderstanding, but I am puzzled with this step.

Reply:
Everything is triggered by the subscript operator:
```python
self.spec['name'].libs
```
which returns a wrapper object that act like a spec. I couldn't look at this PR in more details yet, but remember that the default implementation looks for things in prefix, so you need to have things already installed if you want `.libs` to return anything useful.

Reply:
thanks, i am getting somewhere i think... Just need to do some try/catch and should work, i hope...